scipy/stats/_wilcoxon.py

```diff
@@ -97,38 +97,45 @@ def _wilcoxon_iv(x, y, zero_method, correction, alternative, method, axis):
         raise ValueError(message)
 
     if not isinstance(method, stats.PermutationMethod):
-        methods = {"auto", "approx", "exact"}
+        methods = {"auto", "asymptotic", "exact"}
         message = (f"`method` must be one of {methods} or "
                    "an instance of `stats.PermutationMethod`.")
         if method not in methods:
             raise ValueError(message)
-    output_z = True if method == 'approx' else False
+    output_z = True if method == 'asymptotic' else False
```

Comment:
what I noted: if method=auto and we switch to asymptotic later, we do not output z. I am not sure if this was intended

Reply:
Yes, this is intended. When there are NaNs and `nan_policy='omit'`, the number elements in each slice may vary. Consequently, the exact method may be used for some slices and the asymptotic approximation for others. Returning `z` for some but not others causes problems. (One could argue that `z` _should_ always be included in the result object, but that is a different question.)